server/src/exporter.js

```diff
@@ -0,0 +1,255 @@
+      return;
+    }
+    let toCheck = files.filter((file) => file.startsWith("export--"));
+    let count = toCheck.length;
+    let deletedCount = 0;
+    let errorCount = 0;
+    function oneDone(deleted, errored) {
+      count--;
+      if (deleted) {
+        deletedCount++;
+      }
+      if (errored) {
+        errorCount++;
+      }
+      if (count <= 0) {
+        console.info("EXPORT: finished cleanup run;", toCheck.length, "checked;", deletedCount, "deleted;", errorCount, "had errors; total time:", (Date.now() - start) + "ms");
+      }
+    }
+    for (let dir of toCheck) {
+      let fullDir = path.join(config.exportBase, dir);
+      let file = path.join(fullDir, "pageshot-export.zip");
+      fs.stat(file, (error, stat) => {
+        if (error && error.code == "ENOENT") {
+          fs.stat(fullDir, (error, stat) => {
+            if (error && error.code == "ENOENT") {
+              console.info("EXPORT: directory", fullDir, "disappeared during cleanup");
+              oneDone(false, false);
+              return;
+            } else if (error) {
+              console.warn("EXPORT: could not stat directory", fullDir, ":", error);
+              oneDone(false, true);
+              return;
+            }
+            if (stat.mtime.getTime() + keepTime < Date.now()) {
+              rimraf(fullDir, (error) => {
+                if (error) {
+                  oneDone(false, true);
+                  console.warn("EXPORT: failed to remove unfinished export", fullDir, ":", error);
+                } else {
+                  oneDone(true, false);
+                  console.info("EXPORT: removed unfinished export", fullDir);
+                }
+              });
+            }
+          });
+          return;
+        } else if (error) {
+          oneDone(false, true);
+          console.warn("EXPORT: Error doing stat on", file, ":", error);
+          return;
+        }
+        if (stat.mtime.getTime() + keepTime < Date.now()) {
+          rimraf(fullDir, (error) => {
+            if (error) {
+              oneDone(false, true);
+              console.warn("EXPORT: failed to remove outdated export", fullDir, ":", error);
+            } else {
+              oneDone(true, false);
+              console.info("EXPORT: removed outdated export:", fullDir);
+            }
+          });
+        } else {
+          oneDone(false, false);
+        }
+      });
+    }
+  });
+};
+
+exports.setup = function (app) {
+
+  app.get("/export", function (req, res) {
+    if (! req.deviceId) {
+      return require("./server").simpleResponse(res, "You must have the addon installed to export your shots", 403);
+    }
+    require("./views/export").render(req, res);
+  });
+
+  app.post("/export", function (req, res) {
+    if (! req.deviceId) {
+      return require("./server").simpleResponse(res, "You must have the addon installed to export your shots", 403);
+    }
+    launchWget(req.deviceId).then(() => {
```

Comment:
On second thought, since launchWget is async and nothing cares about it's completion, I think you can just make launchWget not return a promise at all but just fire off the async subprocess. The code that calls getWgetStatus will then take care of monitoring for the subprocess' completion.